addons/udes_stock/models/stock_picking.py

```diff
@@ -116,15 +116,21 @@ def _compute_picking_quantities(self):
             has_discrepancies = False
 
             for move in picking.move_lines.filtered(lambda ml: ml.state != "cancel"):
+                # The move quantity done is in the unit of measure of the move not the product
+                # So use the move UoM not the product for what is done and what is left to do.
+                # This avoids the issue where it says left to do is different to the number of
+                # boxes.
+                # NOTE: Different UoMs across a picking could make these computed fields unreliable
+                # in terms of how much work is done/left todo. Example 50% done - moved one crane
+                # but 1000000 screws in one box.
```

Comment:
this note will be useful in 2 years when clients come across this exact scenario and wonder why bob got lumped with moving 100000 screws and it only counted as 1 :laughing: